src/trio/testing/_raises_group.py

```diff
@@ -247,38 +251,16 @@ def __str__(self) -> str:
         if (match := self.match) is not None:
             # If no flags were specified, discard the redundant re.compile() here.
             reqs.append(
-                f"match={match.pattern if match.flags == _regex_no_flags else match!r}",
+                f"match={match.pattern if match.flags == _REGEX_NO_FLAGS else match!r}",
             )
         if self.check is not None:
             reqs.append(f"check={self.check!r}")
         return f'Matcher({", ".join(reqs)})'
 
 
-# typing this has been somewhat of a nightmare, with the primary difficulty making
-# the return type of __enter__ correct. Ideally it would function like this
-# with RaisesGroup(RaisesGroup(ValueError)) as excinfo:
-#   ...
-# assert_type(excinfo.value, ExceptionGroup[ExceptionGroup[ValueError]])
-# in addition to all the simple cases, but getting all the way to the above seems maybe
-# impossible. The type being RaisesGroup[RaisesGroup[ValueError]] is probably also fine,
-# as long as I add fake properties corresponding to the properties of exceptiongroup. But
-# I had trouble with it handling recursive cases properly.
-
-# Current solution settles on the above giving BaseExceptionGroup[RaisesGroup[ValueError]], and it not
-# being a type error to do `with RaisesGroup(ValueError()): ...` - but that will error on runtime.
-
-# We lie to type checkers that we inherit, so excinfo.value and sub-exceptiongroups can be treated as ExceptionGroups
-if TYPE_CHECKING:
-    SuperClass = BaseExceptionGroup
-else:
-    # At runtime, use a redundant Generic base class which effectively gets ignored.
-    SuperClass = Generic
-
-
 @final
 class RaisesGroup(
-    AbstractContextManager[ExceptionInfo[BaseExceptionGroup[E]]],
-    SuperClass[E],
+    Generic[BaseExcT_co],
 ):
```

Comment:
I removed it as part of initial cleaning up, since it wasn't necessary. I suspect it would interfere with overloads on the enter/exit, though I haven't checked.